examples/chip-tool/server.cpp

```diff
@@ -0,0 +1,173 @@
+    if (clusterId != CHIP_ZCL_CLUSTER_ON_OFF)
+    {
+        println("Unknown cluster ID: %d", clusterId);
+        return;
+    }
+
+    if (attributeId != CHIP_ZCL_CLUSTER_ON_OFF_SERVER_ATTRIBUTE_ON_OFF)
+    {
+        println("Unknown attribute ID: %d", attributeId);
+        return;
+    }
+
+    // At this point we can assume that value points to a boolean value.
+    println("OnOff: %d", value[0]);
+}
+}
+
+namespace {
+class ServerCallback : public SecureSessionMgrCallback
+{
+public:
+    virtual void OnMessageReceived(const MessageHeader & header, Transport::PeerConnectionState * state,
+                                   System::PacketBuffer * buffer, SecureSessionMgrBase * mgr)
+    {
+        const size_t data_len = buffer->DataLength();
+        char src_addr[PeerAddress::kMaxToStringSize];
+
+        // as soon as a client connects, assume it is connected
+        VerifyOrExit(buffer != NULL, println("Received data but couldn't process it..."));
+        VerifyOrExit(header.GetSourceNodeId().HasValue(), println("Unknown source for received message"));
+
+        VerifyOrExit(state->GetPeerNodeId() != kUndefinedNodeId, println("Unknown source for received message"));
+
+        state->GetPeerAddress().ToString(src_addr, sizeof(src_addr));
+
+        println("Packet received from %s: %zu bytes", src_addr, static_cast<size_t>(data_len));
+
+        HandleDataModelMessage(buffer);
+        buffer = NULL;
+
+    exit:
+        // SendTo calls Free on the buffer without an AddRef, if SendTo was not called, free the buffer.
+        if (buffer != NULL)
+        {
+            System::PacketBuffer::Free(buffer);
+        }
+    }
+
+    virtual void OnNewConnection(Transport::PeerConnectionState * state, SecureSessionMgrBase * mgr)
+    {
+        CHIP_ERROR err;
+
+        println("Received a new connection.");
+
+        err = state->GetSecureSession().TemporaryManualKeyExchange(remote_public_key, sizeof(remote_public_key), local_private_key,
+                                                                   sizeof(local_private_key));
+        VerifyOrExit(err == CHIP_NO_ERROR, println("Failed to setup encryption\n"));
+
+    exit:
+        return;
+    }
+
+private:
+    /**
+     * Handle a message that should be processed via our data model processing
+     * codepath.
+     *
+     * @param [in] buffer The buffer holding the message.  This function guarantees
+     *                    that it will free the buffer before returning.
+     */
+    void HandleDataModelMessage(System::PacketBuffer * buffer)
+    {
+        ChipZclStatus_t zclStatus = chipZclProcessIncoming((ChipZclBuffer_t *) buffer);
+        if (zclStatus == CHIP_ZCL_STATUS_SUCCESS)
+        {
+            println("Data model processing success!");
+        }
+        else
+        {
+            println("Data model processing failure: %d", zclStatus);
+        }
+        System::PacketBuffer::Free(buffer);
+    }
+};
+
+static ServerCallback gCallbacks;
```

Comment:
static not needed in anonymous namespace

Reply:
Fixed